Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -13,11 +13,98 @@
 #include <AzCore/Console/IConsole.h>
 #include <AzCore/Math/MathReflection.h>
 #include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/Debug/ProfilerBus.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryScriptUtils.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/optional.h>
+#include <AzCore/IO/Path/Path.h>
 
 #include <AzFramework/Components/ConsoleBus.h>
+#include <AzFramework/IO/LocalFileIO.h>
+#include <AzFramework/Windowing/NativeWindow.h>
+
+#include <Atom/RPI.Public/Pass/AttachmentReadback.h>
+#include <Atom/RPI.Public/RPISystemInterface.h>
 
 namespace ScriptAutomation
 {
+    namespace Utils
+    {
+        bool SupportsResizeClientAreaOfDefaultWindow()
+        {
+            return AzFramework::NativeWindow::SupportsClientAreaResizeOfDefaultWindow();
+        }
+
+        void ResizeClientArea(uint32_t width, uint32_t height)
+        {
+            AzFramework::NativeWindowHandle windowHandle = nullptr;
+            AzFramework::WindowSystemRequestBus::BroadcastResult(
+                windowHandle,
+                &AzFramework::WindowSystemRequestBus::Events::GetDefaultWindowHandle);
+
+            AzFramework::WindowSize clientAreaSize = {width, height};
```

Comment:
do we care about range checking? do we want to vet these values for non-zero and place a limit of reasonable resources like 8K resolution?
I guess you handle it with the AZ_Error below

Reply:
This behavior is copied from ASV, it possible new issues will crop up but I'd like to wait to see if we use the feature and we hit issues.